### ISSUE TITLE
[BUG] Getting "Execution context was destroyed, most likely because of a navigation" when clearing session

### ISSUE THREAD

**Issue by Mark-ICS** [microsoft/playwright#27406]
### System info
- Playwright Version: 1.38.1
- Operating System: macOS 13.5.2
- Browser: Chromium
- Other info:

### Source code

```typescript
await page.click('"Logout"');  # NOTE: This should redirect users to login page on a different domain
await page.evaluate(() => window.sessionStorage.clear());
```

- [ ] I provided exact source code that allows reproducing the issue locally.

**Config file**

```js
import { PlaywrightTestConfig } from "@playwright/test";

const config: PlaywrightTestConfig = {
  projects: [
    {
      name: "Chrome Stable",
      use: {
        browserName: "chromium",
        channel: "chrome",
        screenshot: "only-on-failure",
      },
    },
  ],
  retries: 3,
};
export default config;

```

**Steps**
 - Run the test

**Expected**

 - Test to pass

**Actual**

I get the following error on the line `await page.evaluate(() => window.sessionStorage.clear());`:

```
Error: page.evaluate: Execution context was destroyed, most likely because of a navigation
```

This has recently stopped working, however I'm unable to pinpoint the exact version is broke.

Basically, when I logout, it redirects to a different domain. Then I try to clear the session storage for that domain, and I get the exception mentioned above.

**Comment by Mark-ICS** [comment]
Sorry I should add that this happens intermittently. So about 50% of the time.

**Comment by pavelfeldman** [comment]
I would think that the following happens.

- You start from a page navigated to url `A`.
- You click, which schedules the navigation to url `B`. Navigations in web are async, actions `schedule` the navigation, not perform it.
- While navigation is in progress, you evaluate in the current execution context `A`, which is destroyed under your feet.

You should make sure that the navigation is done. The best way of doing this is to assert some text on the screen, for example:

```js
await expect(page.getByText('You have been signed out')).toBeVisible();
```

After that, it is safe to evaluate. Note that the sessionStorage is origin-specific, so you will have different sessionStorages depending on whether you are currently on origin matching `A` or `B`.

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.

**Comment by Mark-ICS** [comment]
Thank you for responding @pavelfeldman, that's really helpful and it solved the issue. Thanks so much 🙏
